packages/menu/src/MenuItem.ts

```diff
@@ -82,42 +80,73 @@ export class MenuItem extends ActionButton {
         `;
     }
 
-    protected firstUpdated(changes: PropertyValues): void {
+    protected async firstUpdated(changes: PropertyValues) {
         this.setAttribute('tabindex', '-1');
         super.firstUpdated(changes);
         if (!this.hasAttribute('id')) {
             this.id = `sp-menu-item-${MenuItem.instanceCount++}`;
         }
+
+        // Slot updates happens after the connected callback,
+        // so we need to wait a frame before announcing ourselves
+        // or the right menu might not pick this up
+        await new Promise((ready) => requestAnimationFrame(ready));
```

Comment:
This is the big thing that was tripping me up before. Curious if this feels like the right approach.

Reply:
This says that _every_ item should wait, what does the item really care of this? Should the menu that listens do the waiting?

You are already doing a version of this with the debounce in `Menu.updateMenuItems()`. If it were extended in include a `slotchange` resolution, could we get await without needing special handling here?

Reply:
What I like about the event here is that it simplifies the ownership issues. The menu item sends out a "hey I'm here" event, and the right parent picks it up (but only if we do this `await` because of the timing in `sp-menu-group`).

We could also manage this in the parent menu, but we'd need to either need to have each child menu item "trigger" a similar event to determine ownership, or walk the DOM/sub-menus to determine it.

If it weren't for the `await` here I think it'd be a no brainer.

Reply:
I always forget about the menu in the menu group…🤦‍♂️

Reply:
We need to make sure to add `sp-menu-group` (assuming we keep it) to the comment here or this is the sort of things that would get refactored without fully knowing what's going on here!